pkg/ottl/ottlfuncs/func_formattime.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type FormatTimeArguments[K any] struct {
+	Time   ottl.TimeGetter[K]
+	Format string
+}
+
+func NewFormatTimeFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("FormatTime", &FormatTimeArguments[K]{}, createFormatTimeFunction[K])
+}
+
+func createFormatTimeFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*FormatTimeArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("FormatTimeFactory args must be of type *FormatTimeArguments[K]")
```

Comment:
```suggestion
		return nil, errors.New("FormatTimeFactory args must be of type *FormatTimeArguments[K]")
```
Nitpick, but we don't need `fmt.Errorf` here since the entire string is static.